app/services/single_logout_handler.rb

```diff
@@ -52,6 +52,22 @@ def identity
     end
   end
 
+  def user
+    current_user || sp_slo_user
+  end
+
+  def sp_slo_user
+    sp_slo_identity&.user
+  end
+
+  def sp_slo_identity
+    @_sp_slo_identity ||= Identity.includes(:user).find_by(uuid: name_id)
+  end
+
+  def name_id
+    saml_request&.name_id
+  end
```

Comment:
These 4 methods are basically duplicating what `SamlIdpLogoutConcern` already does. What do you think about defining `user` there, and passing in `user` instead of `current_user` here: https://github.com/18F/identity-idp/blob/master/app/controllers/concerns/saml_idp_logout_concern.rb#L10